src/Workleap.DomainEventPropagation.Analyzers/EventDomainAttributeUsageAnalyzer.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Workleap.DomainEventPropagation.Analyzers.Internals;
+using Workleap.Extensions.MediatR.Analyzers.Internals;
+
+namespace Workleap.DomainEventPropagation.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class EventDomainAttributeUsageAnalyzer : DiagnosticAnalyzer
+{
+    internal static readonly DiagnosticDescriptor UseDomainEventAttribute = new(
+        id: RuleIdentifiers.UseDomainEventAttribute,
+        title: "Use DomainEvent attribute on event",
+        messageFormat: "Use the DomainEvent attribute",
+        category: RuleCategories.Design,
+        defaultSeverity: DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
+        helpLinkUri: RuleIdentifiers.HelpUri);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+        UseDomainEventAttribute);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.RegisterCompilationStartAction(OnCompilationStarted);
+    }
+
+    private static void OnCompilationStarted(CompilationStartAnalysisContext context)
+    {
+        var analyzer = new AnalyzerImplementation(context.Compilation);
+        if (analyzer.IsValid)
+        {
+            context.RegisterSymbolAction(analyzer.AnalyzeOperationInvocation, SymbolKind.NamedType);
+        }
+    }
+
+    private sealed class AnalyzerImplementation
+    {
+        private readonly INamedTypeSymbol? _domainEventType;
+        private readonly INamedTypeSymbol? _domainEventAttributeType;
+        private readonly ImmutableHashSet<INamedTypeSymbol> _domainEventTypes;
```

Comment:
_domainEventTypes doesn't seem to be useful in this analyzer, we already individually null check the other fields in `IsValid`